### PR TITLE
[master] MetaData can now directly define a decorator

### PR DESCRIPTION
**Pull request by tommydeboer**

### CODE REVIEW

File: molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java [hunk 1]
```diff
@@ -234,7 +237,24 @@ public Repository add(EntityMetaData emd, RepositoryDecoratorFactory decoratorFa
...
+		if (emd.getDecorator() == null)
+		{
+			return add(emd, new NonDecoratingRepositoryDecoratorFactory());
+		}
+		else
+		{
+			return add(emd, new RepositoryDecoratorFactory()
+			{
+				@Override
+				public Repository createDecoratedRepository(Repository repository)
```
Comment (dennishendriksen) [inline molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java]: Can you add unit test for addEntityMeta for the case when no or one decorator is available?

File: molgenis-data/src/main/java/org/molgenis/data/support/AbstractEntityMetaData.java [hunk 2]
```diff
@@ -321,4 +324,16 @@ public AttributeMetaData getAttribute(String attributeName)
...
 		return attr;
 	}
+
+	@Override
+	public RepositoryDecorator getDecorator()
+	{
+		return decorator;
+	}
+
+	public AbstractEntityMetaData setDecorator(RepositoryDecorator decorator)
```
Comment (dennishendriksen) [inline molgenis-data/src/main/java/org/molgenis/data/support/AbstractEntityMetaData.java]: How to get a handle to the decorated repository in a EntityMetaData class?

File: molgenis-data/src/main/java/org/molgenis/data/RepositoryDecorator.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package org.molgenis.data;
+
+public interface RepositoryDecorator extends Repository
```
Comment (dennishendriksen) [inline molgenis-data/src/main/java/org/molgenis/data/RepositoryDecorator.java]: Please add AbstractRepositoryDecorator that by default forwards all requests to the decorated repository. Add protected Repository getDecoratedRepository() that returns the decorated repository. Rename setTarget to setDecoratedRepository()

**Comment by Mark-de-Haan** [comment]
@tommydeboer Please review comments

**Comment by tommydeboer** [comment]
Haven't found time for this yet since it is a not a story on the board. I'll look at it ASAP.

**Comment by tommydeboer** [comment]
The current solution only works with a fresh app (dropped database, etc.). Any restart will cause MOLGENIS to forget the decorators.

Due to lack of time, I am closing this for now. I'll be back.

**Pull request closed without merging**